version: '3.8'

x-shared-api-worker-env: &shared-api-worker-env
  API_DEBUG: ${API_DEBUG:-0}
  API_DEBUG_PORT: ${API_DEBUG_PORT:-5678}
  CACHE_DIR: ${CACHE_DIR:-/app/volumes}
  JAEGER_LOG_API: ${JAEGER_LOG_API:-http://jaeger:4318/v1/traces}

services:
  api:
    image: contract-review-api
    volumes:
      - ./volumes:/app/volumes
    working_dir: /app
    command: ./api/run.sh
    restart: always
    environment:
      <<: *shared-api-worker-env
    ports:
      - ${API_DEBUG_EXPOSED_PORT}:${API_DEBUG_PORT:-5678}
    build:
      context: ../
      dockerfile: ./api/Dockerfile

  nginx:
    image: nginx:latest
    ports:
      - "8143:8143"
    volumes:
      - ../nginx/conf.d:/etc/nginx/conf.d
      - ../nginx/ssl:/etc/nginx/ssl
    restart: always

  jaeger:
    image: jaegertracing/jaeger:2.7.0
    ports:
      - "16686:16686" # web ui port
      # - "4318:4318" internal log endpoint
    volumes:
      - ../jaeger/config.yaml:/jaeger/config.yaml
      # have to create ./volumes/jaeger and chmod 777 it before starting jaeger
      - ./volumes/jaeger:/home/jaeger
    command: --config /jaeger/config.yaml

networks:
  ai-networks: